enos/modules/test_cluster_health/scripts/versions.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+set -euo pipefail
+
+error_exit() {
+    printf 'Error: %s' "${1}" 
+    exit 1
+}
+
+# Servers version
+server_versions=$(nomad server members -json | jq -r '[.[] | select(.Status == "alive") | .Tags.build] | unique')
+
+if [ "$(echo "$server_versions" | jq 'length')" -eq 0 ]; then
+    error_exit "Unable to get servers version"
+fi
+
+if [ "$(echo "$server_versions" | jq 'length')" -ne 1 ]; then
+    error_exit "Servers are running different versions: $(echo "$server_versions" | jq -c '.')"
+fi
+
+final_version=$(echo "$server_versions" | jq -r '.[0]'| xargs)
+SERVERS_VERSION=$(echo "$SERVERS_VERSION" | xargs)
+
+if [ "$final_version" != "$SERVERS_VERSION" ]; then
+    error_exit "Servers are not running the correct version. Found: $final_version, Expected: $SERVERS_VERSION"
+fi
+
+echo "All servers are running Nomad version $SERVERS_VERSION"
+
+# Clients version
```

Comment:
When we get a little further along, we might want to break this up into two scripts so that we can upgrade servers, run some tests with the mismatched server/client versions, and then upgrade the clients. But this LGTM for now.

Reply:
That is why I added separate variables for client version and server version, se me can run it after upgrading the servers, with old clients and then when everything is upgraded.

Reply:
Oh, I completely misread that! :+1: